tiledb/sm/config/config.cc

```diff
@@ -719,6 +715,23 @@ void Config::inherit(const Config& config) {
   }
 }
 
+Status Config::set_profile(
+    const std::optional<std::string>& profile_name,
+    const std::optional<std::string>& profile_dir) {
+  // Create a Profile object
+  rest_profile_ = RestProfile(profile_name, profile_dir);
+
+  // Load the Profile
+  try {
+    rest_profile_.value().load_from_file();
+  } catch (const RestProfileException& e) {
+    throw RestProfileException(
```

Comment:
If this path is reached then this method still has the side effect of setting `rest_profile_`.  If the rest of the code assumes that the profile is loaded then this could be trouble.  Better to assign to `rest_profile_` after loading to avoid having this object be in an invalid state.

Reply:
This is a really good comment! Thanks!